src/providers/BrandingService.js

```diff
@@ -49,7 +50,15 @@ export class BrandingService extends EventEmitter {
       return
     }
 
-    this.cssVars = newVars
+    const brandColorVariants = updates.brandColor
```

Comment:
You should move this before line 49.

Reply:
What I meant is to move the entire new block prior of that return, which basically will not emit the event if the newVars are the same as the cssVars. But we actively change `newVars`, so they will be always different of `cssVars`. We need **another way** to protect ourselves from emitting the `change` event.